const i18n = require('i18n');
const notificationRepo = require("../../modules/Notification/notification.repo");
const notificationHelper = require("../../helpers/notification.helper")

exports.createNotification = async (req, res) => {
    try {
        const operationResultObject = await notificationRepo.create(req.body);
        notificationHelper.sendPushNotification("Test", "Test Notification", req.body.deviceTokens)
        return res.status(operationResultObject.code).json(operationResultObject);

    } catch (err) {
        console.log(`err.message controller`, err.message);
        return res.status(500).json({
            success: false,
            code: 500,
            error: i18n.__("internalServerError")
        });
    }

}


exports.listNotifications = async (req, res) => {
    try {
        const filterObject = req.query;
        const pageNumber = req.query.page || 1, limitNumber = req.query.limit || 10
        const operationResultObject = await notificationRepo.list(filterObject, {}, { timestamp: -1 }, pageNumber, limitNumber);
        return res.status(operationResultObject.code).json(operationResultObject);

    } catch (err) {
        console.log(`err.message`, err.message);
        return res.status(500).json({
            success: false,
            code: 500,
            error: i18n.__("internalServerError")
        });
    }
}


exports.getNotification = async (req, res) => {
    try {
        const filterObject = req.query;
        const operationResultObject = await notificationRepo.get(filterObject, {});
        return res.status(operationResultObject.code).json(operationResultObject);

    } catch (err) {
        console.log(`err.message`, err.message);
        return res.status(500).json({
            success: false,
            code: 500,
            error: i18n.__("internalServerError")
        });
    }
}


exports.updateNotification = async (req, res) => {
    try {
        const operationResultObject = await notificationRepo.update(req.query._id, req.body);
        return res.status(operationResultObject.code).json(operationResultObject);

    } catch (err) {
        console.log(`err.message`, err.message);
        return res.status(500).json({
            success: false,
            code: 500,
            error: i18n.__("internalServerError")
        });
    }
}


exports.removeNotification = async (req, res) => {
    try {
        const operationResultObject = await notificationRepo.remove(req.query._id);
        return res.status(operationResultObject.code).json(operationResultObject);
    } catch (err) {
        console.log(`err.message`, err.message);
        return res.status(500).json({
            success: false,
            code: 500,
            error: i18n.__("internalServerError")
        });
    }
}
